python/cudf/cudf/core/groupby/groupby.py

```diff
@@ -760,6 +870,22 @@ def shift(self, periods=1, freq=None, axis=0, fill_value=None):
         result = self._groupby.shift(Table(value_columns), periods, fill_value)
         return self.obj.__class__._from_table(result)
 
+    def _mimic_pandas_order(
```

Comment:
If `_mimic_pandas_order` is expected to return DataFrame or Series does it mean that `fillna` might have chances of returning `DataFrame or Series` ?

Or the typing should have been `) -> DataFrame:`

Reply:
Hmm, it should work with Series too. But when I'm testing it with Series, some `RuntimeError` occurs. Investigating.

Reply:
Made a couple updates to `grouping`. Mainly to handle cases with specified with different `by` objects. The goal is to filter out the proper value columns that gets passed down to libcudf. Now `fillna` should handle `DataFrame` and `Series` just as fine.